pkg/v1beta1/cache/cache.go

```diff
@@ -0,0 +1,166 @@
+package cache
+
+import (
+	"fmt"
+	"strings"
+
+	"sync"
+
+	"github.com/go-logr/logr"
+	v1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	klog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// GatewayLookupCache provides concurrency safe lookups from dns host to namespace/gateway
+// Use New() as the Add, Delete, and Get functions all assume the cache map is non-nil
+type GatewayLookupCache struct {
```

Comment:
nit: This could be an internal type and `New()` could return an interface, removing the need for the comment.

Reply:
I am not sure that that interface would be. It would likely be a number of them, including the k8s event handler funcs to make this useable the way that it is on the controller currently.